testing/src/main/kotlin/org/neo4j/connectors/kafka/testing/assertions/TopicVerifier.kt

```diff
@@ -91,11 +109,74 @@ class TopicVerifier<K, V>(
     }
   }
 
+  @Suppress("UNCHECKED_CAST")
+  private fun <V> convert(
+      converter: Converter,
+      assertionClass: Class<V>,
+      topic: String,
+      value: ByteArray?
+  ): Any? {
+    return when (val sourceValue = converter.toConnectData(topic, value).value()) {
+      is Struct ->
+          when (assertionClass) {
+            ChangeEvent::class.java -> sourceValue.toChangeEvent()
+            Map::class.java -> structToMap(sourceValue)
+            else -> sourceValue as V
+          }
+      else -> sourceValue
+    }
+  }
+
+  private fun structToMap(struct: Struct): Map<String, Any?> {
+    val map = mutableMapOf<String, Any?>()
+    struct
+        .schema()
+        .fields()
+        .filter { struct.get(it) != null }
+        .forEach { field ->
+          when (field.schema().type()) {
+            Schema.Type.STRUCT -> map[field.name()] = structToMap(struct.getStruct(field.name()))
+            Schema.Type.ARRAY -> map[field.name()] = convertList(struct.getArray<Any>(field.name()))
+            else -> map[field.name()] = struct.get(field)
+          }
+        }
+    return map
```

Comment:
Could we make use of DynamicTypes in here?

Reply:
yes, it's better with DynamicTypes. Thanks!